cache.go

```diff
@@ -552,6 +552,19 @@ func (c *Cache[K, V]) OnEviction(fn func(context.Context, EvictionReason, *Item[
 	}
 }
 
+// Range iterate over all items and calls fn function. It calls fn function
+// until it returns false.
+func (c *Cache[K, V]) Range(fn func(item *Item[K, V]) bool) {
+	for item := c.items.lru.Front(); item != c.items.lru.Back().Next(); item = item.Next() {
+		c.items.mu.RLock()
```

Comment:
This still has a pretty big problem: `c.items.lru` is left unprotected by the mutex. I think we need to lock it before we enter the loop and relock it before the next iteration starts.

Reply:
You are right. I missed that point. I fixed it in b5ee285.